gnat2goto/driver/tree_walk.adb

```diff
@@ -1130,6 +1138,118 @@ package body Tree_Walk is
       end if;
    end Do_Full_Type_Declaration;
 
+   ---------------------
+   -- Name_Has_Prefix --
+   ---------------------
+
+   function Name_Has_Prefix (N : Node_Id; Prefix : String) return Boolean
+     with Pre => Present (Name (N));
+
+   function Name_Has_Prefix (N : Node_Id; Prefix : String) return Boolean is
+      Short_Name : constant String :=
+        Get_Name_String (Chars (Name (N)));
+   begin
+      if Prefix'Length > Short_Name'Length then
```

Comment:
it would be slightly less verbose to write

`return Prefix'Length <= Short_Name'Length and then Prefix = Short_Name (...);`

but it is a matter of style (though in general, Ada is already quite verbose, so it makes sense to write concisely)